db/cryptodata/decrypter.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cryptodata
+
+import (
+	"encoding/json"
+	"strings"
+	"encoding/base64"
+)
+
+// DecryptFunc is function that decrypts arbitrary data
+type DecryptFunc func(inData []byte) (data [] byte, err error)
+
+// ValidateFunc is used to validate compatibility of JSON data when decrypting
+type ValidateFunc func(inData []byte) (isValid bool)
```

Comment:
This naming doesn't fit here. It doesn't validate the JSON data, it just checks if the data contains encrypted data so it can start looking for it.

Perhaps `CheckEncryptedFunc`?